### PR TITLE
fix(@rspack/core): minify should be disabled when minimizer is an empty array

### LINKED ISSUE #4899: should not minify code when optimization.minimizer= []
### System Info

  System:
    OS: macOS 13.4.1
    CPU: (10) arm64 Apple M1 Pro
    Memory: 99.73 MB / 32.00 GB
    Shell: 5.9 - /bin/zsh
  Binaries:
    Yarn: 3.2.2 - ~/Library/Application Support/fnm/node-versions/v16.17.1/installation/bin/yarn
    npm: 7.24.2 - ~/Library/Application Support/fnm/node-versions/v16.17.1/installation/bin/npm
  Browsers:
    Chrome: 119.0.6045.199
    Safari: 16.5.1
  npmPackages:
    @rspack/cli: latest => 0.4.1 
    @rspack/core: latest => 0.4.1 

### Details

should not minify code when optimization.minimizer = []

### Reproduce link

https://github.com/web-infra-dev/rspack-repro/tree/minify-bug

### Reproduce Steps

pnpm install & pnpm build

### PR DESCRIPTION
**Pull request by xc2**
## Summary

fixes #4899

## Test Plan

- [x] new test case added: `packages/rspack/tests/configCases/plugins/minify-disabled-on-empty-array`
- [x] all test cases with name match `/minify/` should pass

## Require Documentation?

- [x] No
- [ ] Yes, the corresponding rspack-website PR is \_\_

### CODE REVIEW

**Comment by hardfist** [comment]
@xc2 you need to update snapshot

File: packages/rspack/tests/__snapshots__/StatsTestCases.test.ts.snap [hunk 1]
```diff
@@ -544,7 +544,7 @@ exports[`StatsTestCases should print correct stats for filename 1`] = `
   "errors": [],
   "errorsCount": 0,
   "filteredModules": undefined,
-  "hash": "d11c74c648d060ebcdfd",
+  "hash": "1019e65f759482cee6e5",
```
Comment (hardfist) [inline packages/rspack/tests/__snapshots__/StatsTestCases.test.ts.snap:547]: @xc2 can you help whey the hash changed here? did the minify logic change for this case?

Comment (xc2) [reply packages/rspack/tests/__snapshots__/StatsTestCases.test.ts.snap:547]: still checking, might respond later tonight

- [x] what cases? looks like all simple cases with dynamic import, thus these hash changes might be `splitChunks` related or dynamic import related.
- [x] does the building artifacts (case `dist`) change? no
- [x] does the module order change? maybe no per the stats output.
- [ ] actual reason? checking, guess it being the plugins order change

Comment (hardfist) [reply packages/rspack/tests/__snapshots__/StatsTestCases.test.ts.snap:547]: @xc2 do you find the reason?

**Comment by ahabhgk** [comment]
!eco-ci

**Review by ahabhgk — APPROVED** [review]
Actually `builtin.minifyOptions` is legacy debt, I can't even remember what the origin behavior should be, but since it will be drop support in 0.5, so let's make the behavior make sense for SwcJsMinimizerPlugin, and try my best to compatible with `builtin.minifyOptions`

I changed a bit on top of your PR. Thanks for your contribution!

**Comment by xc2** [comment]
thanks @ahabhgk

**Pull request merged**